src/controller/java/src/chip/devicecontroller/model/AttributeWriteRequest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package chip.devicecontroller.model;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+/** An attirbute write request that should be used for interaction model write interactio. */
+public class AttributeWriteRequest {
+  private ChipPathId endpointId, clusterId, attributeId;
+  private Optional<Integer> dataVersion = Optional.empty();
+  private byte[] tlv;
+
+  private AttributeWriteRequest(
+      ChipPathId endpointId,
+      ChipPathId clusterId,
+      ChipPathId attributeId,
+      byte[] tlv,
+      Optional<Integer> dataVersion) {
+    this.endpointId = endpointId;
+    this.clusterId = clusterId;
+    this.attributeId = attributeId;
+    this.tlv = tlv;
+    this.dataVersion = dataVersion;
+  }
+
+  public ChipPathId getEndpointId() {
+    return endpointId;
+  }
+
+  public ChipPathId getClusterId() {
+    return clusterId;
+  }
+
+  public ChipPathId getAttributeId() {
+    return attributeId;
+  }
+
+  public int getDataVersion() {
+    return dataVersion.orElse(0);
+  }
+
+  public boolean hasDataVersion() {
+    return dataVersion.isPresent();
+  }
+
+  public byte[] getTlvByteArray() {
+    return tlv;
+  }
+
+  @Override
+  public boolean equals(Object object) {
+    if (object instanceof AttributeWriteRequest) {
```

Comment:
Are we intentionally omitting the TLV data and data version from the equality test?

Reply:
yes, intentional. this would be used for path comparison. updated comment, thanks